threatgrid/ctia
# Fix pagination

> Close #threatgrid/iroh/issues/4192

The inconsistent results in feeds are due to the `all-pages` function that performs a `search_after` based pagination using the default `sort_by` parameter. However this default value is base on `_doc` field which is unique per shard but not per index! Indeed this is a problem that can occur for any `search_after` pagination that will use the default `sort_by` value. 
This PR changes the default `sort_by` value to `_doc,id` that will first sort by `_doc` to ensure the order then by `id` for disambiguation. It also generalizes tests with multiple shards to better reflect production properties.

<a name="qa">[§](#qa)</a> QA
============================

1. Create feed data with a single bulk that contains all data of a "big" feed:
- 1 indicator
- 350 judgements
- 350 relationships linking every created judgments to the created indicator (judgement id as `source_ref` and indicator id as `target_ref`).
2. Create the feed instance for the previous indicator.
3. make 10 calls to corresponding view `/ctia/feed/{feed-id}/view?s={feed-share-token}` and check that the responses are always the same.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================



```
intern: fixed edge case pagination bug.
```
